### PR TITLE
Substrate Updates

### PR DESCRIPTION
**Pull request by AurevoirXavier**
https://github.com/darwinia-network/substrate-update-tracking/issues/12
https://github.com/darwinia-network/substrate-update-tracking/issues/15

### CODE REVIEW

File: bin/node/runtime/pangolin/src/pallets/babe.rs [hunk 1]
```diff
@@ -6,6 +6,8 @@ use sp_core::crypto::KeyTypeId;
 use crate::*;
 
 frame_support::parameter_types! {
+	// NOTE: Currently it is not possible to change the epoch duration after the chain has started.
+	//       Attempting to do so will brick block production.
```
Comment (AsceticBear) [inline bin/node/runtime/pangolin/src/pallets/babe.rs:10]: break?

File: bin/node/runtime/pangolin/src/lib.rs [hunk 2]
```diff
@@ -42,7 +42,11 @@ pub mod constants {
 	pub const DAYS: BlockNumber = 24 * HOURS;
 
 	pub const MILLISECS_PER_BLOCK: Moment = 6000;
+	// NOTE: Currently it is not possible to change the slot duration after the chain has started.
+	//       Attempting to do so will brick block production.
```
Comment (AsceticBear) [inline bin/node/runtime/pangolin/src/lib.rs:46]: break?

File: bin/node/runtime/pangolin/src/lib.rs [hunk 3]
```diff
@@ -42,7 +42,11 @@ pub mod constants {
...
 	pub const SLOT_DURATION: Moment = MILLISECS_PER_BLOCK;
+	// NOTE: Currently it is not possible to change the epoch duration after the chain has started.
+	//       Attempting to do so will brick block production.
```
Comment (AsceticBear) [inline bin/node/runtime/pangolin/src/lib.rs:49]: break?

File: bin/node/runtime/pangolin/src/pallets/babe.rs (hunk 1 above)
Comment (AurevoirXavier) [reply bin/node/runtime/pangolin/src/pallets/babe.rs:10]: But I didn't change it

File: frame/elections-phragmen/src/lib.rs [hunk 4]
```diff
@@ -1345,12 +1449,11 @@ mod tests {
 	}
 
 	fn has_lock(who: &u64) -> u64 {
-		dbg!(Balances::locks(who));
 		Balances::locks(who)
 			.get(0)
 			.cloned()
 			.map(|lock| {
-				assert_eq!(lock.id, ElectionsPhragmenModuleId::get());
+				assert_eq!(lock.id, PhragmenElectionPalletId::get());
```
Comment (AsceticBear) [inline frame/elections-phragmen/src/lib.rs:1456]: Why not `ElectionsPhragmenPalletId`?

Comment (AurevoirXavier) [reply frame/elections-phragmen/src/lib.rs:1456]: companion for #polkadot-2765

File: bin/node/runtime/pangolin/src/pallets/proxy.rs [hunk 5]
```diff
@@ -41,7 +41,7 @@ impl InstanceFilter<Call> for ProxyType {
 							Call::Democracy(..) |
 							Call::Council(..) |
 							Call::TechnicalCommittee(..) |
-							Call::ElectionsPhragmen(..) |
+							Call::PhragmenElection(..) |
```
[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply bin/node/runtime/pangolin/src/pallets/proxy.rs:44]: companion for #polkadot-2765

**Review by AsceticBear — APPROVED** [review]

**Review by HackFisher — APPROVED** [review]
In polkadot's companion release, it is requesting a mandatory upgrade for node operators, I'm not sure whether we need same request for nodes if using the same substrate dependency, or it is OK for us since polkadot have other changes in client side? (Host, parachain?)

Ref: https://github.com/paritytech/polkadot/releases/tag/v0.9.0

**Comment by HackFisher** [comment]
```
Notes for Validators:

In order to support parachains, private or rotating IP addresses will not work, you must have a public static IP address for your validator
The flag --no-private-ipv4 would disable the node from searching for private IP addresses (like you might on a local / dev chain). This flag's behaviour is now the default
A validator node must have inbound p2p ports open. Your node will not be able to participate in validating parachains if it does not have this.
```

Are these required to darwinia clients(pangolin, crab, darwinia)?

**Comment by AurevoirXavier** [comment]
> In polkadot's companion release, it is requesting a mandatory upgrade for node operators, I'm not sure whether we need same request for nodes if using the same substrate dependency, or it is OK for us since polkadot have other changes in client side? (Host, parachain?)
> 
> Ref: https://github.com/paritytech/polkadot/releases/tag/v0.9.0

I don't think this's a mandatory upgrade for us. But need some test on Pangolin.

**Comment by AurevoirXavier** [comment]
> In polkadot's companion release, it is requesting a mandatory upgrade for node operators, I'm not sure whether we need same request for nodes if using the same substrate dependency, or it is OK for us since polkadot have other changes in client side? (Host, parachain?)
> 
> Ref: https://github.com/paritytech/polkadot/releases/tag/v0.9.0

We allow private address on our network. https://github.com/darwinia-network/darwinia-common/pull/604/commits/532c8904b71c6a475f98c38435e14ad6c837de11

**Pull request merged**